File: app/src/main/java/lt/tesonet/model/ServerDao.java
package lt.tesonet.model;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import io.reactivex.Single;

import static android.arch.persistence.room.OnConflictStrategy.REPLACE;

@Dao
public interface ServerDao {

    @Insert(onConflict = REPLACE)
    void addAll(List<Server> servers);

    @Query("SELECT * FROM server")
    Single<List<Server>> getAllServers();

    @Query("DELETE FROM server")
    void clearServers();
}

File: Assigment/app/src/main/java/assignment/tesonet/homework/storage/database/dao/ServerDao.kt
package assignment.tesonet.homework.storage.database.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert

import android.arch.persistence.room.OnConflictStrategy.REPLACE
import android.arch.persistence.room.Query
import android.support.annotation.WorkerThread
import assignment.tesonet.homework.api.response.Server

@Dao
interface ServerDao {
    @WorkerThread
    @Insert(onConflict = REPLACE)
    @JvmSuppressWildcards
    fun save(list: List<Server>)

    @WorkerThread
    @Query("SELECT * FROM server")
    fun load(): List<Server>

    @WorkerThread
    @Query("DELETE FROM server")
    fun delete()
}